app.js

```diff
@@ -50,20 +48,26 @@ Date.prototype.getWeekNo = function(){
 	return Math.ceil((((d-new Date(d.getFullYear(),0,1))/8.64e7)+1)/7);
 };
 
-// This is basically an empty shell function
-// It could be used for document pre- or post- handling
-function insertdb(response) {
-	//console.log('--- INSERTDB: statusCode: ', response.statusCode);
-	response.on('error', function(e) {
-		console.log('--- INSERT_DB: ')
-		console.error(e);
-	});
-	response.on('end', function() {
-		if (response.statusCode === 409) {
-			console.log('--- INSERT_DB: document already exists.');
-		}
-	});
-}
+var https = require('https');
+var timer = null;
+var stack = [];
+
+function process() {
+   var item = stack.shift();
+   console.log("process: " + item.opts.path);
+   https.request(item.opts, item.func).end();
+   if (stack.length === 0) {
+       clearInterval(timer);
+       timer = null;
+   };
+ };
+
+function throttle(item) {
+    stack.push(item);
+    if (timer === null) {
+      timer = setInterval(process, config.interval);
+    };
+};
```

Comment:
Throttle class isn't working for me, temporarily added this snippet back in so I could continue with development.